torch/_tensor.py

```diff
@@ -205,6 +205,9 @@ def storage(self):
         if self.dtype not in torch.storage._dtype_to_storage_type_map():
             raise RuntimeError(f'unsupported Storage type: {self.dtype}')
 
+        if self.device.type not in ['cuda', 'cpu']:
+            raise RuntimeError(f'cannot create storage for device: {self.device}')
```

Comment:
We shouldn't actually have this requirement, right? Because part of the reason we are doing this refactor is to uniformly treat storage over all device types.